public/app/plugins/datasource/grafana-azure-monitor-datasource/utils/migrateQuery.test.ts

```diff
@@ -115,4 +115,103 @@ describe('AzureMonitor: migrateQuery', () => {
+          }),
+        })
+      );
+    });
+    it('correctly updates old filter containing wildcard', () => {
+      const dimensionFilters: AzureMetricDimension[] = [{ dimension: 'TestDimension', operator: 'eq', filter: '*' }];
+      const result = migrateQuery({ ...azureMonitorQueryV8, azureMonitor: { dimensionFilters } });
+      expect(result).toMatchObject(
+        expect.objectContaining({
+          azureMonitor: expect.objectContaining({
+            dimensionFilters: [
+              { dimension: dimensionFilters[0].dimension, operator: dimensionFilters[0].operator, filters: ['*'] },
+            ],
+          }),
+        })
+      );
+    });
+    it('correctly updates old filter containing value', () => {
+      const dimensionFilters: AzureMetricDimension[] = [{ dimension: 'TestDimension', operator: 'eq', filter: 'test' }];
+      const result = migrateQuery({ ...azureMonitorQueryV8, azureMonitor: { dimensionFilters } });
+      expect(result).toMatchObject(
+        expect.objectContaining({
+          azureMonitor: expect.objectContaining({
+            dimensionFilters: [
+              { dimension: dimensionFilters[0].dimension, operator: dimensionFilters[0].operator, filters: ['test'] },
+            ],
+          }),
+        })
+      );
+    });
+    it('correctly ignores wildcard if filters has a value', () => {
+      const dimensionFilters: AzureMetricDimension[] = [
+        { dimension: 'TestDimension', operator: 'eq', filter: '*', filters: ['testFilter'] },
+      ];
+      const result = migrateQuery({ ...azureMonitorQueryV8, azureMonitor: { dimensionFilters } });
+      expect(result).toMatchObject(
+        expect.objectContaining({
+          azureMonitor: expect.objectContaining({
+            dimensionFilters: [
+              {
+                dimension: dimensionFilters[0].dimension,
+                operator: dimensionFilters[0].operator,
+                filters: ['testFilter'],
+              },
+            ],
+          }),
+        })
+      );
+    });
+    it('correctly ignores duplicates', () => {
+      const dimensionFilters: AzureMetricDimension[] = [
+        { dimension: 'TestDimension', operator: 'eq', filter: 'testFilter', filters: ['testFilter'] },
+      ];
+      const result = migrateQuery({ ...azureMonitorQueryV8, azureMonitor: { dimensionFilters } });
+      expect(result).toMatchObject(
+        expect.objectContaining({
+          azureMonitor: expect.objectContaining({
+            dimensionFilters: [
+              {
+                dimension: dimensionFilters[0].dimension,
+                operator: dimensionFilters[0].operator,
+                filters: ['testFilter'],
+              },
+            ],
+          }),
+        })
+      );
+    });
+    it('correctly merges values', () => {
+      const dimensionFilters: AzureMetricDimension[] = [
+        { dimension: 'TestDimension', operator: 'eq', filter: 'testFilter2', filters: ['testFilter'] },
+      ];
```

Comment:
I think that the behavior should be to ignore `filter` if `filters` is set. There should not be a case in which both are used, queries should be either migrated / new or non-migrated.